crates/rust-analyzer/src/conv.rs

```diff
@@ -302,6 +303,76 @@ impl ConvWith<&FoldConvCtx<'_>> for Fold {
     }
 }
 
+impl Conv for &'static str {
+    type Output = (SemanticTokenType, Vec<SemanticTokenModifier>);
+
+    fn conv(self) -> (SemanticTokenType, Vec<SemanticTokenModifier>) {
```

Comment:
We could do a more direct conversion here, avoiding the `Vec` allocation, right?

Reply:
I think `smallvec` could be more appropriate here?

Reply:
The problem is using `&'static str` type in the first place. It doesn't make sense to optimize impl details while the API itself is wrong. I think we should move to a model closer to `tag+modifier` of LSP.